swxsoc/util/tests/test_util.py

```diff
@@ -417,3 +421,310 @@ def test_create_configdir_configured(instrument, time, level, version, result):
     del os.environ["SWXSOC_CONFIGDIR"]
     swxsoc._reconfigure()
 # fmt: on
+
+
+@mock_aws
+def test_search_all_attr():
+    conn = boto3.resource("s3", region_name="us-east-1")
+
+    bucket_name = "swxsoc-eea"
+    conn.create_bucket(Bucket=bucket_name)
+
+    s3 = boto3.client("s3")
+    s3.put_object(
+        Bucket=bucket_name,
+        Key="l0/2024/04/swxsoc_EEA_l0_2024094-124603_v01.bin",
+        Body=b"test data 1",
+    )
+    s3.put_object(
+        Bucket=bucket_name,
+        Key=f"l1/2024/04/swxsoc_eea_l1_{time_formatted}_v1.2.3.cdf",
+        Body=b"test data 2",
+    )
```

Comment:
Depending if you want to reuse some of these resources you can create a pytest fixture so you don't have to re-create for each test case.

Reply:
I'll try to improve this later on ^